code/modules/reagents/chemistry/reagents/drink_reagents.dm

```diff
@@ -209,6 +210,35 @@
 		holder.remove_reagent(/datum/reagent/consumable/capsaicin, 2)
 	..()
 
+/*See block comment in ../milk/overdose_process(mob/living/M) for calculation and explanation of why this exists and why 5 was chosen
+* For best results use in tandem with method outlined in this comment
+*/
+/datum/reagent/consumable/milk/overdose_start(mob/living/M)
+	M.reagents.add_reagent(/datum/reagent/toxin/bonehurtingjuice, 5) //The integer here should match var/starting_amount in ../milk/overdose_process(mob/living/M)
+	return ..()
+
+/datum/reagent/consumable/milk/overdose_process(mob/living/M)
+	var/minimum_cycles, amount_to_add //minimum_cycles is the number of ticks for an amount of units equal to the overdose threshold to process. amount_to_add is the calculated amount to add per tick to meet ensure that target_units after minimum_cycle ticks.
+	var/datum/reagent/converted_reagent = /datum/reagent/toxin/bonehurtingjuice //Needed to get the metabolism for desired reagent, exists solely for brevity compared to /datum/reagent/category/reagent.metabolization_rate
+	minimum_cycles = overdose_threshold/metabolization_rate
+	var/target_units = 50 //Target amount of converted_reagent to exist in system after overdose_threshold units have processed
+	var/starting_amount = 5 //Should match the integer in ../milk/overdose_start(mob/living/M)
+	var/converted_metabolism = converted_reagent.metabolization_rate //Extra brevity with converted reagent metabolism
+	amount_to_add = ((target_units-starting_amount)/minimum_cycles)+converted_metabolism
+	M.reagents.add_reagent(/datum/reagent/toxin/bonehurtingjuice, amount_to_add)
+	return ..()
```

Comment:
Aaaaa, try to avoid using variables like this., target units and starting amount are _always_ the same value. Making it a bit of a waste of memory to make them variables.

My last comment meant making it adaptable to different metabolic rates, everything else isn't needed though 😆

Reply:
Ok I can hard code the starting amount and target units

Reply:
I mean it's a thing that exists and can be used in other places, which is arguably a benefit.